### PR TITLE
PayPal: Verify method_supports refunds in API response

### PR DESCRIPTION
**Pull request by timmyc**
This branch seeks to fix https://github.com/Automattic/wp-calypso/issues/21665

__The Problem__
Currently the Store UI in Calypso shows the refund button as an option for orders where payment was remitted via PayPal Standard. When one attempts to process the refund, the process fails because no API user/pass/signature has been setup on the payment method. We don't expose the option for setting these values within Store, but they are available in wp-admin.

The proposed solution in this branch is to modify the `method_supports` array that is returned from the `wc/v3/payment_gateways` endpoint. When the the PayPal gateway configuration is missing any/all of the api credentials ( `api_username`, `api_password`, `api_signature` ), the API response will omit `refunds` in the `method_supports` array. If all items are present/not empty strings, the `method_supports` will include `products` and `refunds`. While we don't verify the API credentials are correct, this will at least not signal to API-driven apps like Store on WordPress.com that a refund can be processed via the API for PayPal.

Calypso then uses this `method_supports` data to display either the refund button, or show the manual refund verbiage:

__Refunds with API Creds in Calypso__
refund-via-paypal

__Refund Dialog with NO API Creds__
manual-refund

__To Test__
- Run the `phpunit` test suite to verify the new tests for this are passing
- Use postman and issue GET requests to `/wc/v3/payment_gateways`. First make a request with no api credentials set, verify `method_supports` only shows `products` for PayPal

method-supports-no-api-keys

- Then visit `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=paypal` and add in some jibberish to the API Credentials section

api-credentials

- Re-do the GET request to `/wc/v3/payment_gateways` again, and note that `refunds` are now supported:

refunds-supported

### CODE REVIEW

**Comment by timmyc** [comment]
> I wonder if it would be better to do the button hiding logic on the UI side instead?

For sure we could totally do this, and that is where I first started to fix the issue - but then saw how we were using `method_supports` to determine _if_ a refund could be performed for the given payment method gateway... so that is what led me here.

> Previously, method_supports matches up with the static supports list from each gateway. I.e. what a gateway CAN support, and not what it is actively supporting based on configuration.

Interesting... I didn't know the backstory of `method_supports`. It seems the way we are using it in Calypso kind of goes against that. Almost like there needs to be both `method_supports` and `methods_enabled` or something. I'm guessing PayPal might be a unique case in this regard where one can accept payments without full API keys.

> If we want to update the API this way, should we do this change in wc-api-dev instead?

Yeah that is probably a better spot, especially if we wanted this to get into core.

**Comment by justinshreve** [comment]
>  Almost like there needs to be both method_supports and methods_enabled or something.

I definitely see the value in returning what's possible/enabled (so the clients using the API can easily consume it) -- so I like this idea!

**Comment by justinshreve** [comment]
@timmyc It looks like core does want to fix this the same way (modify the supports array) https://github.com/woocommerce/woocommerce/pull/19380/files -- so the way you do this should be fine. We could ship this as a hotfix until `3.4.0` is released.

**Comment by timmyc** [comment]
Rebased @justinshreve - so are you cool with this getting merged in?

**Comment by justinshreve** [comment]
@timmyc I'll do a proper code review & test tomorrow morning for you so I can leave an approval (I don't think I fully finished reviewing it before leaving my comment last week) -- but yeah I'm cool with us doing this 👍.

File: inc/wc-calypso-bridge-paypal-method-supports.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	}
+
+	if ( 'paypal' === $gateway->id ) {
+        $api_username = $gateway->get_option( 'api_username' );
+        $api_password = $gateway->get_option( 'api_password' );
+        $api_signature = $gateway->get_option( 'api_signature' );
+
+        // If api username, password or signature is not set, i.e. empty string, we can't support woo-based refunds
+        if ( empty( $api_username ) || empty( $api_password ) || empty( $api_signature ) ) {
+            $response->data[ 'method_supports' ] = array(
```
[the remark this answers is not in the document]
Comment (timmyc) [reply inc/wc-calypso-bridge-paypal-method-supports.php]: Great point, I'll update the logic to do just that.

**Comment by timmyc** [comment]
@justinshreve updated the items you listed above.

**Pull request merged**